src/engraving/rendering/dev/accidentalslayout.cpp

```diff
@@ -0,0 +1,1112 @@
+ *
+ * Copyright (C) 2023 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include <cfloat>
+
+#include "accidentalslayout.h"
+#include "layoutcontext.h"
+#include "../paddingtable.h"
+#include "shape.h"
+#include "tlayout.h"
+
+#include "dom/accidental.h"
+#include "dom/chord.h"
+#include "dom/ledgerline.h"
+#include "dom/note.h"
+#include "dom/score.h"
+
+using namespace mu::engraving;
+using namespace mu::engraving::rendering::dev;
+
+void AccidentalsLayout::layoutAccidentals(const std::vector<Chord*>& chords, LayoutContext& ctx)
+{
+    std::vector<Accidental*> allAccidentals = collectAccidentals(chords);
+
+    if (allAccidentals.empty()) {
+        return;
+    }
+
+    for (Accidental* acc : allAccidentals) {
+        acc->computeMag();
+        TLayout::layoutAccidental(acc, acc->mutldata(), ctx.conf());
+    }
+
+    AccidentalPlacementEngine accidentalPlacementEngine(allAccidentals, chords);
+    accidentalPlacementEngine.doAccidentalPlacement();
+}
+
+std::vector<Accidental*> AccidentalsLayout::collectAccidentals(const std::vector<Chord*> chords)
+{
+    std::vector<Accidental*> accidentals;
+
+    for (Chord* chord : chords) {
+        for (Note* note : chord->notes()) {
+            Accidental* acc = note->accidental();
+            if (!acc) {
+                continue;
+            }
+            acc->setPos(0.0, 0.0);
+            bool isRedundant = false;
+            for (Accidental* otherAcc : accidentals) {
+                if (otherAcc->accidentalType() == acc->accidentalType() && otherAcc->line() == acc->line()) {
+                    isRedundant = true;
+                    break;
+                }
+            }
+            if (isRedundant) {
+                acc->setbbox(RectF());
+                continue;
+            }
+            accidentals.push_back(acc);
+        }
+    }
+
+    return accidentals;
+}
+
+AccidentalsLayout::AccidentalPlacementEngine::AccidentalPlacementEngine(const std::vector<Accidental*>& allAccidentals,
+                                                                        const std::vector<Chord*>& chords)
+    : m_allAccidentals(allAccidentals), m_chords(chords)
+{
+    sortTopDown(m_allAccidentals);
+
+    Accidental* accidental = allAccidentals.front();
+    const MStyle& style = accidental->style();
+
+    m_spatium = accidental->spatium();
+
+    m_verticalAccidentalToAccidentalClearance = 0.15 * m_spatium;
```

Comment:
magic numbers

Reply:
This 0.15 (and other magic numbers) is used in a lot of different places in this PR. Worth adding a named constexpr

Reply:
I've removed all the magic numbers I had used around the files and collected them all here with proper names and a comment that explains that they are just engraving parameters that don't have a style. The fact that some of them have the same value (like 0.15) is just a coincidence, they actually all mean different things.